### PR TITLE
Assorted fixes

### PR DESCRIPTION
**Pull request by parport0**
We have discovered a couple of places where the return values of functions weren't checked. Add the missing checks, and add `tags` and `TAGS` to .gitignore while at it.

These issues were found with SVACE static analysis tool.

### CODE REVIEW

File: tee-supplicant/src/teec_ta_load.c [hunk 1]
```diff
@@ -138,7 +138,11 @@ static int try_load_secure_module(const char* prefix,
 	}
 
 	s = ftell(file);
-	if (s > *ta_size || !ta) {
+	if (s == (size_t) -1) {
```
Comment (jforissier) [inline tee-supplicant/src/teec_ta_load.c:141]: `if (s < 0)` would be better

**Comment by jforissier** [comment]
For commit: ".gitignore: add tags and TAGS":
`Reviewed-by: Jerome Forissier <jerome@forissier.org>`

File: tee-supplicant/src/teec_ta_load.c (hunk 1 above)
Comment (parport0) [reply tee-supplicant/src/teec_ta_load.c:141]: s is size_t. Also, ftell always returns -1 on error.

Comment (jenswi-linaro) [reply tee-supplicant/src/teec_ta_load.c:141]: Hmm, wouldn't it be better to use the correct type when taking the return value from `ftell()`?

File: tee-supplicant/src/teec_ta_load.c [hunk 2]
```diff
@@ -138,7 +138,11 @@ static int try_load_secure_module(const char* prefix,
...
+		printf("failed to ftell the TA file\n");
```
Comment (jenswi-linaro) [inline tee-supplicant/src/teec_ta_load.c:142]: We're printing the name of the file above on error, please do so here too.

File: tee-supplicant/src/teec_ta_load.c (hunk 1 above)
Comment (parport0) [reply tee-supplicant/src/teec_ta_load.c:141]: It's probably more of a matter of taste; alright, I'll add a new `long` variable.

File: tee-supplicant/src/teec_ta_load.c [hunk 3]
```diff
@@ -137,7 +138,14 @@ static int try_load_secure_module(const char* prefix,
...
 
-	s = ftell(file);
+	l = ftell(file);
+	if (l == -1) {
+		DMSG("failed to ftell the ta %s TA-file", fname);
+		fclose(file);
+		return TA_BINARY_NOT_FOUND;
+	}
+
+	s = (size_t) l;
```
Comment (jenswi-linaro) [inline tee-supplicant/src/teec_ta_load.c:148]: Cast not needed?

File: tee-supplicant/src/teec_ta_load.c [hunk 4]
```diff
@@ -137,7 +138,14 @@ static int try_load_secure_module(const char* prefix,
 		return TA_BINARY_NOT_FOUND;
 	}
 
-	s = ftell(file);
+	l = ftell(file);
+	if (l == -1) {
```
Comment (jenswi-linaro) [inline tee-supplicant/src/teec_ta_load.c:142]: We should perhaps rule out all negative values here since we're going to put this in an unsigned variable further down.

File: tee-supplicant/src/teec_ta_load.c [hunk 5]
```diff
@@ -157,7 +161,7 @@ static int try_load_secure_module(const char* prefix,
 	}
 
 	if (s != fread(ta, 1, s, file)) {
-		printf("error fread TA file\n");
+		DMSG("failed to fread the ta %s TA-file", fname);
```
Comment (jforissier) [inline tee-supplicant/src/teec_ta_load.c:164]: Changing `printf()` to `DMSG()` looks like a good idea but it should be a separate commit IMO (note there is another instance in `try_load_secure_module()`)

**Review by jforissier — COMMENTED** [review]
With my comment below addressed:
`Reviewed-by: Jerome Forissier <jerome@forissier.org>`
